app/controllers/v0/sessions_controller.rb

```diff
@@ -47,13 +47,15 @@ def new
             end
       render json: { url: url }
     end
-    # rubocop:enable Metrics/CyclomaticComplexity, Metrics/MethodLength
+    # rubocop:enable Metrics/CyclomaticComplexity
 
     def logout
-      session = Session.find(Base64.urlsafe_decode64(params[:session]))
-      raise Common::Exceptions::Forbidden, detail: 'Invalid request' if session.nil?
-      destroy_user_session!(User.find(session.uuid), session)
-      redirect_to SAML::URLService.new(saml_settings, session: session, user: current_user).slo_url
+      session_object = Session.find(Base64.urlsafe_decode64(params[:session]))
+      raise Common::Exceptions::Forbidden, detail: 'Invalid request' if session_object.nil?
+      @session_object = session_object
+      @current_user = User.find(session_object.uuid)
+      reset_session
+      redirect_to SAML::URLService.new(saml_settings, session: session_object, user: current_user).slo_url
     end
```

Comment:
this is one of the first things i'm looking forward to getting rid of shortly after launch. 😂